v4.1.8 migration- Add secrets migration information (#851)
strapi/documentation: docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.0.x-to-4.1.8.md

  },
});

```

</code-block>

<code-block title="TYPESCRIPT">

```jsx
//path: config/admin.ts

export default ({ env }) => ({
  auth: {
    secret: env('ADMIN_JWT_SECRET'),
  },
  apiToken: {
    salt: env('API_TOKEN_SALT'),
  },
});

```

</code-block>

</code-group>

## Configuring `JWT_SECRET`

`JWT_SECRET` is used by the Users and Permissions plugin, and populated in `/.env`. The property should be stored in `config/plugins.js` (or `config/plugins.ts` for a TypeScript project). The `plugins` file is not created by default in a Strapi application. If the file does not exist, users should create the file and add the follow code snippet.

<code-group>

<code-block title="JAVASCRIPT">

```jsx
//  path: config/plugins.js

module.exports = ({ env }) => ({
  // ...
  'users-permissions': {
    config: {
    jwtSecret: env('JWT_SECRET')
  },
  },
  // ...
});

```

</code-block>

<code-block title="TYPESCRIPT">

```jsx
//  path: config/plugins.ts

export default ({ env }) => ({
  // ...
  'users-permissions': {
    config: {
    jwtSecret: env('JWT_SECRET')
  },
  },
  // ...
});

```

</code-block>

</code-group>

## Setting secrets for non-development environments

> Users are required to set secrets for each unique environment, such as a prodcution environment deployment on a platform. Strapi no longer passes the following secrets to non-development environments:

Reviewer: ```suggestion
Users are required to set secrets for each unique environment, such as a production environment deployment on a platform. Strapi no longer passes the following secrets to non-development environments:
```

- APP_KEYS
- JWT_SECRET
- API_TOKEN_SALT
- ADMIN_JWT_SECRET

> There are multiple methods to generate secrets, such as running `openssl rand -base64 32` in the terminal (Mac and Linux OS). Generating unique secrets for each environment is recommended for increased security.

Reviewer: ```suggestion
There are multiple methods to generate secrets, for example running `openssl rand -base64 32` in the terminal (Mac and Linux OS). Generating unique secrets for each environment is recommended for increased security.
```

---
title: Migrate from 4.1.7 to 4.1.8 - Strapi Developer Docs
description: Learn how you can migrate your Strapi application from 4.1.7 to 4.1.8.
canonicalUrl: https://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.0.x-to-4.1.8.html
---

# v4.0.x to v4.1.8 migration guide

> The Strapi v4.0.x to v4.1.8 migration guide upgrades versions of v4.0.6 through v4.1.7 to v4.1.8. The minimum configuration for `config/admin` now includes the API token `API_TOKEN_SALT`. Additionally, Strapi no long populates default values for the admin JSON web token in `config/admin`. Initial values are generated and stored in the .env file during project creation. The migration to v4.1.8 consists of 2 steps:

Reviewer: You can just use `admin JWT` I think, no need to spell it out entirely.

I think it would also be better to say "Strapi no longer generates random values for secrets in environments other than development"

Reviewer: Can you throw something on my calendar for Thursday and we can step through this one together? (I have free time tmrw, Wednesday too but not sure if you wanted to stay that late)

---
title: Migrate from 4.1.7 to 4.1.8 - Strapi Developer Docs
description: Learn how you can migrate your Strapi application from 4.1.7 to 4.1.8.
canonicalUrl: https://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.0.x-to-4.1.8.html
---

# v4.0.x to v4.1.8 migration guide

> The Strapi v4.0.x to v4.1.8 migration guide upgrades versions of v4.0.6 through v4.1.7 to v4.1.8. The minimum configuration for `config/admin` now includes the API token `API_TOKEN_SALT`. Additionally, Strapi no long populates default values for the admin JWT in `config/admin`. Initial values are generated and stored in the .env file during project creation. Strapi no longer passes secrets to non-developmen environments, requiring users to set the secrets purposefully. The migration to v4.1.8 consists of 3 steps:

Reviewer 2: Simple "noob" question from a non-native English speaker:
Should it be "no long" or "no longer" (populates default values…)? 🤔

Reviewer 2: (also I think we're missing a "t", "non-developmen" environments 😄

Reviewer 3: Yes, it should be "longer". I think it was a copy and paste error or the delicate equilibrium of caffeine.
